app/controllers/prison/visits_controller.rb

```diff
@@ -27,23 +27,28 @@ def update
   end
 
   def nomis_cancelled
-    visit = scoped_visit
+    visit = scoped_visit.find(params[:id])
     visit.confirm_nomis_cancelled
     flash[:notice] = t('nomis_cancellation_confirmed', scope: [:prison, :flash])
     redirect_to prison_inbox_path
   end
 
   def deprecated_show
-    @visit = unscoped_visit
+    @visit = unscoped_visit.find(params[:id])
   end
 
   def show
-    @visit = scoped_visit
-    @message = @visit.messages.first
+    @visit = scoped_visit.
+             includes(
+               :visitors,
+               messages: :user,
+               visit_state_changes: :processed_by).
+             find(params[:id])
```

Comment:
That's a lot of SQL/ActiveRecord for a controller. Perhaps a presenter layer would improve testability and readability of this controller.